Add USE_SYSTEM options for KLEIDI, CUDNN_FRONTEND, CUTLASS, and FMT

sladyn98 opened the pull request:
This commit adds CMake options to allow users to use system-installed versions of four libraries instead of the bundled versions
Fixes #153863

Skylion007 commented on cmake/Summary.cmake:
```diff
@@ -81,7 +81,9 @@ function(caffe2_print_configuration_summary)
       message(STATUS "    cuDNN version       : ${CUDNN_VERSION}")
+      message(STATUS "    USE_SYSTEM_CUDNN_FRONTEND : ${USE_SYSTEM_CUDNN_FRONTEND}")
     endif()
+    message(STATUS "    USE_SYSTEM_CUTLASS  : ${USE_SYSTEM_CUTLASS}")
```
Likewise haven't done a great job of maintaining backward compatibility here for changes since it was previously pinned.

Skylion007 commented on cmake/Dependencies.cmake:
```diff
@@ -1572,22 +1595,30 @@ endif()
+  list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
+  set(BUILD_SHARED_LIBS ${TEMP_BUILD_SHARED_LIBS} CACHE BOOL "Build shared libs" FORCE)
+else()
+  find_package(fmt REQUIRED)
```
Should still use fmt install "ON" or the headers will be missing from then you wheel includes and it will break compiling some downstream torch extensions (unless those CMake files are update too).

Skylion007 reviewed (changes requested):
Would dangerously break portability of the torch wheel as is. Fmtlib version should ideally match PyTorch's so extensions should just grab the include in the wheel. They could be overrided to use system lib too, but whats the benefit of versioning fmtlib from a system install anyway (CUDA stuff I could at least make some argument in Docker level CUTLASS/CUDNN changes.

sladyn98 committed:
Add USE_SYSTEM options for KLEIDI, CUDNN_FRONTEND, CUTLASS, and FMT

Adds CMake options to allow using system-provided versions of libraries
instead of bundled versions:
- USE_SYSTEM_KLEIDIAI
- USE_SYSTEM_CUDNN_FRONTEND
- USE_SYSTEM_CUTLASS
- USE_SYSTEM_FMT

All options are enabled automatically when USE_SYSTEM_LIBS=ON.

Key features:
- fmt headers always installed to wheel for extension compatibility
- Version detection and warnings for CUDNN_FRONTEND and CUTLASS
- Alerts users about known SDPA dispatch bugs in old CUDNN_FRONTEND
- Warns about backward compatibility issues in CUTLASS (previously pinned)
- Use at your own risk messaging for system library versions

Addresses reviewer feedback from PR #166824.
